### ISSUE TITLE
GraphQL requests does not work from Apollo on React Native

### ISSUE THREAD

**Issue by sangdth** [nhost/cli#130]
Current I have issue while using `@nhost/react-auth` and `@nhost/react-apollo` in an Expo (React Native) project, still not sure but based on the error message, I give it a try here.

The wrapper:
```typescript
import { NhostAuthProvider } from '@nhost/react-auth';
import { NhostApolloProvider } from '@nhost/react-apollo';
...
      <NhostAuthProvider nhost={nhost}>
        <NhostApolloProvider nhost={nhost}>
        ...
              <RootStack />
        ...
        </NhostApolloProvider>
      </NhostAuthProvider>
```

The nhost:
```typescript
export const nhost = new NhostClient({
  backendUrl: 'http://localhost:1337',
  clientStorage: SecureStore,
  clientStorageType: 'expo-secure-storage',
});
```

The query:
```typescript
import gql from 'graphql-tag';
...
export const JOBS_NEARBY_AGGREGATE = gql`
  query JOBS_NEARBY_AGGREGATE (
    $args: jobs_nearby_args!,
    $limit: Int!,
    $offset: Int!,
    $where: jobs_bool_exp!
  ) {
    jobs_nearby_aggregate(args: $args, limit: $limit, offset: $offset, where: $where) {
      __typename
      nodes {
        id
        created_at
        description
        expiration_date
        quantity
        title
        updated_at
        address {
          id
          unstructured_value
          location
        }
      }
    }
  }
`;
```

The implementation:
```typescript
  const { data, error } = useQuery(JOBS_NEARBY_AGGREGATE, {
    variables: {
      args: {
        distance: 9999,
        user_id: user?.id ?? '', // correct id
      },
      limit: 10,
      offset: 0,
      where: {
        company: {
          tenant_id: { _eq: '...' }, // correct id
        },
      },
    },
    fetchPolicy: 'network-only',
    errorPolicy: 'all',
  });
```
The issue:
```bash
Response: Bad Request

// Copied as cURL
curl 'http://localhost:1337/v1/graphql' \
  -H 'accept: */*' \
  -H 'Referer: ' \
  -H 'authorization: Bearer correct token here' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 12_0_1) AppleWebKit/537.36 (KHTML, like Gecko) ReactNativeDebugger/0.12.1 Chrome/87.0.4280.141 Electron/11.4.6 Safari/537.36' \
  -H '__RN_DEBUGGER_SET_HEADER_REQUEST_sec-websocket-protocol: graphql-ws' \
  -H 'content-type: application/json' \
  --data-binary $'{"operationName":"JOBS_NEARBY_AGGREGATE","variables":{"args":{"distance":9999,"user_id":"b63259c1-c4b6-450d-9ee7-baf297515570"},"limit":10,"offset":0,"where":{"company":{"tenant_id":{"_eq":"3ad7aaec-1ba8-476c-b378-d1257f212a94"}}}},"query":"query JOBS_NEARBY_AGGREGATE($args: jobs_nearby_args\u0021, $limit: Int\u0021, $offset: Int\u0021, $where: jobs_bool_exp\u0021) {\\n  jobs_nearby_aggregate(\\n    args: $args\\n    limit: $limit\\n    offset: $offset\\n    where: $where\\n  ) {\\n    __typename\\n    nodes {\\n      id\\n      created_at\\n      description\\n      expiration_date\\n      quantity\\n      title\\n      updated_at\\n      address {\\n        id\\n        unstructured_value\\n        location\\n        __typename\\n      }\\n      __typename\\n    }\\n  }\\n}\\n"}' \
  --compressed
```
<img width="1157" alt="Screenshot 2021-12-07 at 19 09 28" src="https://user-images.githubusercontent.com/1083478/145074588-1d1fefd7-9059-436a-9453-173f7710f673.png">

Every time I make a query request from React Native, the nhost backend show this error message:

```
2021/12/07 18:58:21 websocketproxy: couldn't upgrade websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
```
---

To compare, these response below are exactly the same query on Next.js application:
```bash
curl 'http://localhost:1337/v1/graphql' \
  -H 'Connection: keep-alive' \
  -H 'sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="96", "Google Chrome";v="96"' \
  -H 'accept: */*' \
  -H 'content-type: application/json' \
  -H 'authorization: Bearer correct token here' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.55 Safari/537.36' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'Origin: http://localhost:3000' \
  -H 'Sec-Fetch-Site: same-site' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Referer: http://localhost:3000/' \
  -H 'Accept-Language: en-US,en;q=0.9,vi-VN;q=0.8,vi;q=0.7' \
  --data-raw $'{"operationName":"JOBS_NEARBY_AGGREGATE","variables":{"args":{"distance":9999,"user_id":"b63259c1-c4b6-450d-9ee7-baf297515570"},"limit":10,"offset":0,"where":{"company":{"tenant_id":{"_eq":"3ad7aaec-1ba8-476c-b378-d1257f212a94"}}}},"query":"query JOBS_NEARBY_AGGREGATE($args: jobs_nearby_args\u0021, $limit: Int\u0021, $offset: Int\u0021, $where: jobs_bool_exp\u0021) {\\n  jobs_nearby_aggregate(\\n    args: $args\\n    limit: $limit\\n    offset: $offset\\n    where: $where\\n  ) {\\n    __typename\\n    nodes {\\n      id\\n      created_at\\n      description\\n      expiration_date\\n      quantity\\n      title\\n      updated_at\\n      address {\\n        id\\n        unstructured_value\\n        location\\n        __typename\\n      }\\n      __typename\\n    }\\n  }\\n}\\n"}' \
  --compressed
```
Works on Next.js
<img width="1048" alt="Screenshot 2021-12-07 at 19 11 05" src="https://user-images.githubusercontent.com/1083478/145074827-f9a8da9c-56b0-4010-bf20-a6db3d284cfb.png">
<img width="1046" alt="Screenshot 2021-12-07 at 19 11 14" src="https://user-images.githubusercontent.com/1083478/145074836-6bb6aa28-8c1b-423d-8a19-d01283c44110.png">

Works on Postman
<img width="1811" alt="Screenshot 2021-12-07 at 19 12 21" src="https://user-images.githubusercontent.com/1083478/145075002-8f0d9eb1-130a-460a-8f14-1e57a57c49be.png">

CC @elitan

**Comment by elitan** [comment]
We've confirmed that this issue is only present when using the CLI.

When making the same GraphQL request to Nhost directly (`*.nhost.run/v1/graphql`) the GraphQL request works.

**Comment by sangdth** [comment]
It works with remote graphql url:

<img width="959" alt="Screenshot 2021-12-08 at 10 30 26" src="https://user-images.githubusercontent.com/1083478/145175294-d441989a-44e4-49b9-847d-9426dba67ea9.png">

**Comment by elitan** [comment]
https://github.com/nhost/cli/blob/0c5700a1ad059ec47d2e21e5fdf79fc9b5e90926/proxy/proxy.go#L76

Possibly related?

**Comment by mrinalwahal** [comment]
> https://github.com/nhost/cli/blob/0c5700a1ad059ec47d2e21e5fdf79fc9b5e90926/proxy/proxy.go#L76
> 
> Possibly related?

I'm trying to reproduce the error on my end. I'll have a look and update you guys.

Thanks for opening the issue!

**Comment by mrinalwahal** [comment]
This has been fixed in `v0.5.11` onward.
